test/gov/hzn_sdo.sh

```diff
@@ -0,0 +1,1166 @@
+		echo -e "${PREFIX} completed."
+	fi
+else
+	echo -e "${PREFIX} Failed: Wrong error response from 'hzn sdo keys new': exit code: $rc, output: $cmdOutput."
+	exit 1
+fi
+
+# Test hzn sdo keys create
+
+echo -e "${PREFIX} Testing 'hzn sdo keys create' without HZN_EXCHANGE_USER_AUTH set"
+unset HZN_EXCHANGE_USER_AUTH
+cmdOutput=$(hzn sdo keys create /tmp/sdo_key.json 2>&1)
+rc=$?
+if [[ $rc -eq 1 && "$cmdOutput" == *'Error:'*'exchange user authentication must be specified'* ]]; then
+	echo -e "${PREFIX} received expected error response."
+else
+	echo -e "${PREFIX} Failed: Wrong error response from 'hzn sdo keys create' without HZN_EXCHANGE_USER_AUTH set: exit code: $rc, output: $cmdOutput."
+	exit 1
+fi
+export HZN_EXCHANGE_USER_AUTH="$USERDEV_ADMIN_AUTH"
+
+echo -e "${PREFIX} Testing 'hzn sdo keys create' with incorrect HZN_EXCHANGE_USER_AUTH set"
+export HZN_EXCHANGE_USER_AUTH=fakeuser:fakepw
+cmdOutput=$(hzn sdo keys create /tmp/sdo_key.json 2>&1)
+rc=$?
+if [[ $rc -eq 7 && "$cmdOutput" == *'Error:'*'Invalid credentials.'* ]]; then
+	echo -e "${PREFIX} received expected error response."
+else
+	echo -e "${PREFIX} Failed: Wrong error response from 'hzn sdo keys create' with incorrect HZN_EXCHANGE_USER_AUTH set: exit code: $rc, output: $cmdOutput."
+	exit 1
+fi
+export HZN_EXCHANGE_USER_AUTH=$USERDEV_ADMIN_AUTH
+
+echo -e "${PREFIX} Testing 'hzn sdo keys create' without HZN_ORG_ID set"
+unset HZN_ORG_ID
+cmdOutput=$(hzn sdo keys create /tmp/sdo_key.json 2>&1)
+rc=$?
+if [[ $rc -eq 1 && "$cmdOutput" == *'Error:'*'organization ID must be specified'* ]]; then
+	echo -e "${PREFIX} received expected error response."
+else
+	echo -e "${PREFIX} Failed: Wrong error response from 'hzn sdo keys create' with incorrect HZN_ORG_ID set: exit code: $rc, output: $cmdOutput."
+	exit 1
+fi
+export HZN_ORG_ID=$HZN_ORG_ID_SAVE
+
+echo -e "${PREFIX} Testing 'hzn sdo keys create' with incorrect HZN_ORG_ID set"
+export HZN_ORG_ID=fakeorg
+cmdOutput=$(hzn sdo keys create /tmp/sdo_key.json 2>&1)
+rc=$?
+if [[ $rc -eq 7 && "$cmdOutput" == *'Error:'*'Invalid credentials.'* ]]; then
+	echo -e "${PREFIX} received expected error response."
+else
+	echo -e "${PREFIX} Failed: Wrong error response from 'hzn sdo keys create' without HZN_ORG_ID set: exit code: $rc, output: $cmdOutput."
+	exit 1
+fi
+export HZN_ORG_ID=$HZN_ORG_ID_SAVE
+
+echo -e "${PREFIX} Testing 'hzn sdo keys create' without HZN_SDO_SVC_URL set"
+unset HZN_SDO_SVC_URL
+cmdOutput=$(hzn sdo keys create /tmp/sdo_key.json 2>&1)
+rc=$?
+if [[ $rc -eq 7 && "$cmdOutput" == *'Error:'*'Could not get'*'HZN_SDO_SVC_URL'* ]]; then
+	echo -e "${PREFIX} received expected error response."
+else
+	echo -e "${PREFIX} Failed: Wrong error response from 'hzn sdo keys create' without HZN_SDO_SVC_URL set: exit code: $rc, output: $cmdOutput."
+	exit 1
+fi
+export HZN_SDO_SVC_URL=$HZN_SDO_SVC_URL_SAVE
+
+echo -e "${PREFIX} Testing 'hzn sdo keys create <key-file>'"
```

Comment:
Do you need to remove this key before trying to create it, in case the script exited last time after the create and before the remove? (This is only necessary in the e2edev remote case.)

Reply:
The keys are now removed before creation